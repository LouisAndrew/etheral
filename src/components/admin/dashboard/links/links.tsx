import React from 'react';

import { Box } from 'rebass';

import NotionExport from './notion-export';

type Props = {};

const Links: React.FC<Props> = () => {
    return (
        <Box
            height="100vh"
            maxHeight="100vh"
            pb={[5]}
            className="custom-scrollbar"
        >
            <NotionExport />
        </Box>
    );
};

export { Links };
