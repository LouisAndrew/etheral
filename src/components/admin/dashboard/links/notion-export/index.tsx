import React from 'react';

import { html as htmlMain } from './html';
import { html as htmlCMS } from './cms/html';

import './styles.scss';
import './cms/styles.scss';
import './initial-styles.scss';

const cmsNotionExportedSrc =
    'Content%20Management%20System%201319dd58faca4f6eba879a2d34f0db3c';

const templateMain = { __html: htmlMain };
const templateCMS = { __html: htmlCMS.replace(cmsNotionExportedSrc, '') };

const NotionExport: React.FC<{}> = () => (
    <div className="exported">
        <div id="main" dangerouslySetInnerHTML={templateMain} />
        <div id="cms" dangerouslySetInnerHTML={templateCMS} />
    </div>
);

export default NotionExport;
