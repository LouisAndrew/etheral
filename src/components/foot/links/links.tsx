import React from 'react';
import { Link } from '@reach/router';

import { Flex, Text } from 'rebass';

import { theme } from 'styles';

type Props = {};

const Links: React.FC<Props> = () => {
    return (
        <Flex
            as="nav"
            flexDirection={['row']}
            justifyContent="center"
            alignItems="center"
            py={[4, 4, 5]}
            sx={{
                '& > a': {
                    textDecoration: 'none',
                    color: theme.colors.black[0],
                    px: [3],
                },
            }}
        >
            <Link to="/contact">
                <Text variant="linkSmall">CONTACT US</Text>
            </Link>
            <Link to="/">
                <Text variant="linkSmall">FAQ</Text>
            </Link>
            <Link to="/">
                <Text variant="linkSmall">SIZE GUIDE</Text>
            </Link>
        </Flex>
    );
};

export { Links };
