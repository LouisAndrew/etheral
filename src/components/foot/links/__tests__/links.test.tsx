import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Links } from '../links';

describe('Links on footer component', () => {
    const Element = <Links />;

    afterEach(cleanup);

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(Element, div);
    });

    it('matches snapshot', () => {
        const run = true;
        if (run) {
            const tree = renderer.create(Element).toJSON();
            expect(tree).toMatchSnapshot();
        }
    });
});
