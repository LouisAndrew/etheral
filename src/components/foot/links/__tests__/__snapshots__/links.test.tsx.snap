// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links on footer component matches snapshot 1`] = `
<nav
  className="css-16d31i"
>
  <a
    href="/contact"
    onClick={[Function]}
  >
    <div
      className="css-r8xm40"
    >
      CONTACT US
    </div>
  </a>
  <a
    href="/faq"
    onClick={[Function]}
  >
    <div
      className="css-r8xm40"
    >
      FAQ
    </div>
  </a>
  <a
    href="/size-guide"
    onClick={[Function]}
  >
    <div
      className="css-r8xm40"
    >
      SIZE GUIDE
    </div>
  </a>
</nav>
`;
